sdk/core/azure-core/inc/azure/core/http/policy.hpp

```diff
@@ -209,6 +215,98 @@ namespace Azure { namespace Core { namespace Http {
         NextHttpPolicy nextHttpPolicy) const override;
   };
 
+  class ValuePolicy;
+
+  /**
+   * @brief @ValuePolicy options.
+   */
+  class ValuePolicyOptions {
+    friend class ValuePolicy;
+
+  private:
+    std::map<std::string, std::string> m_headerValues;
+    std::map<std::string, std::string> m_queryValues;
```

Comment:
How about just


```
struct ValuePolicyOptions {
    std::map<std::string, std::string> headerValues;
    std::map<std::string, std::string> queryValues;
};
```

Reply:
IMO, not as extensible for future versions, and see the newer version, header keys are case-insensitive. But thank you for the feedback, I would listen to other opinions, would go with the majority.

Reply:
@Jinming-Hu, I did as you suggested, the team was in favor of your proposal. Thank you!

Reply:
just in case you need it, case-insensitive `std::map` https://github.com/Azure/azure-sdk-for-cpp/blob/b3cdb2790c6c44de87fe5aa97c2dc4f917ba5948/sdk/storage/azure-storage-common/inc/azure/storage/common/storage_common.hpp#L69

Reply:
Thank you Jinming, this is great. I didn't know it existed in our repo, but I thought about the idea. So far, we think the simpler the policy is, the less smart job it does - the better, especially given the fact that it is now internal. But if you strongly disagree, I am open to consider to change it.

Reply:
I'm fine with whatever, as long as it works.